### PR TITLE
ISSUE-1168 Support Kafka metrics in SAM

### PR DESCRIPTION
**Pull request by sameer79**
Kafka lag offset graph shows in the  Kafka-Source component node

kafkalagoncomponent

when Kafka-Source componenet is selected, Kafka lag offset graph shown in the bottom panel.

bottompanel

@arunmahadevan  @shahsank3t please review and comment

### CODE REVIEW

File: webservice/src/main/resources/app/scripts/containers/Streams/Metrics/TopologyComponentMetrics.jsx [hunk 1]
```diff
@@ -239,6 +247,20 @@ class TopologyComponentMetrics extends Component {
               {this.state.loadingRecord ? loader : this.getGraph('Queue', queueData, 'step-before', showMetrics)}
             </div>
           </div>
+          {
+            compData.parentType === 'SOURCE' && compData.currentType === "Kafka"
+            ? <div className="component-metric-graph">
+                <div style={{textAlign: "left"}}>kafka Lag Offset</div>
```
Comment (shahsank3t) [inline webservice/src/main/resources/app/scripts/containers/Streams/Metrics/TopologyComponentMetrics.jsx]: nit: k should be in uppercase

**Comment by omkreddy** [comment]
maybe we should rename  "Kafka Lag offset" to "Kafka Consumer Lag" or "Kafka Offset Lag"

**Comment by shahsank3t** [comment]
@omkreddy @arunmahadevan @HeartSaVioR 
http://jsoneditoronline.org/?id=23523623dd16056ad74460824b5822fd
Above is the response we get from the kafka metrics API. 

FYI: on the UI, we are currently using `lag` and `offset` from the response so there will be 2 lines in the graph that can be viewed. 

Let us know if that needs to be changed or whether we should also be using `logsize` from the API response. Thanks.

**Comment by omkreddy** [comment]
In my opinion, "Lag" should be sufficient.

**Comment by arunmahadevan** [comment]
@shahsank3t , the changes looks good. I think lag is more useful and for now that should be sufficient. Can you rename it to "Kafka offset lag" as @omkreddy suggested?

**Comment by sameer79** [comment]
@omkreddy @arunmahadevan @shahsank3t  address the comment and squash commit as well, ready for merge.

**Comment by arunmahadevan** [comment]
+1

**Pull request closed without merging**